hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -1958,15 +1959,17 @@ private void addOMNodeToPeers(String newOMNodeId) throws IOException {
 
         if (newOMNodeDetails == null) {
           // If new node information is not present in the newly loaded
-          // configuration also, throw an exception
+          // configuration also, throw an exception.
+          // This case can also come when we have decommissioned a node and
+          // ratis will apply previous transactions to add that node back.
           throw new IOException("There is no OM configuration for node ID "
               + newOMNodeId + " in ozone-site.xml.");
         }
       }
     } catch (IOException e) {
       LOG.error("{}: Couldn't add OM {} to peer list.", getOMNodeId(),
           newOMNodeId);
-      exitManager.exitSystem(1, e.getLocalizedMessage(), e, LOG);
+      return;
```

Comment:
Doing `return` here will affect all the other `IOExceptions` as well, we should avoid doing this.

Reply:
Done